web/modules/custom/user_timezone/src/UserTimezoneSalutation.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\user_timezone;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Prepares user greeting.
+ */
+class UserTimezoneSalutation {
+  use StringTranslationTrait;
+
+  /**
+   * The current user service.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $currentUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(AccountProxyInterface $currentUser, TimeInterface $time) {
+    $this->currentUser = $currentUser;
+    $this->time = $time;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_user')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSalutation() {
+    $time = $this->time->getRequestTime();
+
+    if ($time >= 06 && (int) $time < 12) {
+      return $this->t('Good morning %username', [
```

Comment:
This code is redundant, you can save morning, afternoon or evening in a variable and add insert it as placeholder (similarly to the user name).

Reply:
I am not sure if that is a good idea. These salutations might be different in certain languages (e.g. good morning vs good afternoon in Japanese is completely different)